src/fsharp/Optimizer.fs

```diff
@@ -1663,7 +1663,15 @@ let TryDetectQueryQuoteAndRun cenv (expr:Expr) =
         //printfn "Not eliminating because no Run found"
         None
                 
-
+let IsSystemStringConcatOverload (methRef: ILMethodRef) =
+    methRef.Name = "Concat" && methRef.DeclaringTypeRef.FullName = "System.String" && 
+    methRef.ArgTypes |> List.forall(fun ilty -> ilty.BasicQualifiedName = "System.String") &&
```

Comment:
Can you please move this check to the end? We now always try to order boolean checks so that collections are checked at the end since the other checks are often faster

Reply:
Sure